src/stepFunctions/wizards/previewStateMachineCDKWizard.ts

```diff
@@ -0,0 +1,235 @@
+import { ConstructNode } from '../../cdk/explorer/nodes/constructNode'
+
+export interface CdkAppLocationPickItem {
+    label: string
+    cdkApplocation: CdkAppLocation | undefined
+}
+
+export interface TopLevelNodePickItem {
+    label: string
+    topLevelNode: ConstructNode | undefined
+}
+
+export interface ConstructNodePickItem {
+    label: string
+    stateMachineNode: ConstructNode | undefined
+}
+
+interface PreviewStateMachineCDKWizardResponse {
+    cdkApplication: CdkAppLocationPickItem
+    topLevelNode: TopLevelNodePickItem
+    stateMachine: ConstructNodePickItem
+}
+
+export class PreviewStateMachineCDKWizard extends MultiStepWizard<PreviewStateMachineCDKWizardResponse> {
+    private cdkApplication?: CdkAppLocationPickItem
+    private topLevelNode?: TopLevelNodePickItem
+    private stateMachine?: ConstructNodePickItem
+    private promptUser: typeof picker.promptUser
+
+    public constructor(promptUser?: typeof picker.promptUser) {
+        super()
+        this.promptUser = promptUser || picker.promptUser.bind(picker)
+    }
+
+    protected get startStep() {
+        return this.SELECT_WORKSPACE_ACTION
+    }
+
+    private readonly SELECT_WORKSPACE_ACTION: WizardStep = async () => {
+        const cdkAppLocationsHelper: CdkAppLocation[] = await detectCdkProjects(vscode.workspace.workspaceFolders)
+        const cdkAppLocations: CdkAppLocationPickItem[] = cdkAppLocationsHelper.map(obj => {
+            return {
+                label: getCDKAppWorkspaceName(obj.cdkJsonPath),
+                cdkApplocation: obj,
+            }
+        })
+
+        if (cdkAppLocations.length === 0) {
+            cdkAppLocations.push({
+                label: localize('Aws.cdk.app.noWorkspace', '[No workspace found]'),
+                cdkApplocation: undefined,
+            })
+        }
+
+        const quickPick = picker.createQuickPick<CdkAppLocationPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.message.prompt.selectCDKWorkspace.placeholder', 'Select CDK workspace'),
+                step: 1,
+                totalSteps: 3,
+            },
+            buttons: [vscode.QuickInputButtons.Back],
+            items: cdkAppLocations,
+        })
+
+        const choices = await this.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        })
+
+        this.cdkApplication = picker.verifySinglePickerOutput<CdkAppLocationPickItem>(choices)
+        return this.cdkApplication ? wizardContinue(this.SELECT_APPLICATION_ACTION) : WIZARD_GOBACK
+    }
+
+    private readonly SELECT_APPLICATION_ACTION: WizardStep = async () => {
+        const appLocation = this.cdkApplication ? this.cdkApplication.cdkApplocation : undefined
+
+        if (!appLocation) {
+            return WIZARD_GOBACK
```

Comment:
I don't think this will have an interactive step, so if a user hits `!appLocation`, they'll automatically be sent back to the last screen with no prompt. I would at least populate a quick pick with a `No App Location` message (or something else) that returns the `WIZARD_GOBACK`.